crypto/trace.c

```diff
@@ -146,19 +146,33 @@ const char *OSSL_trace_get_category_name(int num)
 {
     size_t i;
 
+    /*
+     * Partial check that OSSL_TRACE_CATEGORY_... macros
+     * are synced with trace_categories array
+     */
+    static const size_t trace_categories_len =
+        sizeof(trace_categories)/sizeof(0[trace_categories]);
```

Comment:
There's an OpenSSL-specific macro `OSSL_NELEM()` to get the amount of elements in an array.  You can see it used in the `for` loop a few lines down.

Also, hehehe, it's not often that I see the construct `n[pointer]` used.  Yes, it is legitimate C, but so unusual that it's not what would be called reader friendly code.  While I can appreciate the coolness factor, I'd suggest not doing this in OpenSSL source code, yeah?